### ISSUE TITLE
iSCSI boot-up: No drivers loaded?

### ISSUE THREAD

**Issue by cflachen** [maxnet/berryboot#363]
First of all, thanks for making a great bootloader/installer for the RaspberryPi! I was looking for an easy iSCSI capable loader for a while now.

I am using RaspberryPi 2 in connection with a DROBO b810i.

The initial loading process works just great, where you select the target name and desired Linux OS. But after re-booting, to complete the installation, it won't connect to the iSCSI and goes into the fallback rescue shell.

In this mode, when I mount the memory card onto /mnt and run the customized iscsi setup shell, the system complains that the /lib/modules directory is missing. I was able to manually untar the shared.tgz file into /lib/modules and successfully run the iscsi setup shell.

So it seems like everything would work fine if the proper iscsi modules were available in the initial ramdisk.

Hope this description will help to fix this bug. Let me know if I should try something else or provide additional data.

I used this link for the installation: http://downloads.sourceforge.net/project/berryboot/berryboot-20161118-pi0-pi1-pi2-pi3.zip

The link for only PI2 and PI3 didn't do anything (screen kept black when powering up).

Thanks,
Claus

**Comment by maxnet** [comment]
You do are using a wired Ethernet connection?
Wifi is not supported for iscsi.

>The link for only PI2 and PI3 didn't do anything (screen kept black when powering up).

Make sure the card is formatted FAT32 and not exFAT or NTFS.

Should at least display the rainbow screen.
Can hang after that if you were to accidentally use an unsupported model such as Pi 1.

**Comment by cflachen** [comment]
Update: After going through the source code, I found a hint! In the cmdline.txt file (on the FAT partition of the SDcard), the datadev entry was set to the harddisk partition of the iscsi device, after configuration. In my case it was datadev=sda2.
In the source code however, the special iscsi modules only get loaded when the datadev entry is set to "iscsi". After changing to "datadev=iscsi" in cmdline.txt, the system was booting up properly!
The software first complains that it can't find a system under "iscsi", but at it also has all the extra modules available now. Then it searches all block devices for a proper image and eventually find the correct one.

**Comment by maxnet** [comment]
Did you have an installation that used usb storage previously on same card?

**Comment by cflachen** [comment]
No usb storage was involved. However, I remember that I was trying the "use existing files" option first, which didn't work. Then I selected the now mounted "sda2" partition, which probably made the system think that it is a regular partition and not an iSCSI partition. I bet everything would work if I selected "ext4" to begin with.
I generated an ext3 filesystem on the iSCSI target beforehand, which works better with the Drobo device. By selecting the "existing files" I thought it would use this ext3 partition and not reformat to ext4.
